### DISCUSSION TITLE
Linux: Processes launched from threads block SIGTERM, SIGQUIT and SIGPIPE

### DISCUSSION THREAD

**Discussion by cgudrian** [pocoproject/poco#3792] [General]
On Linux all Poco threads mask the signals TERM, QUIT and PIPE (possibly to ensure that those signals are only delivered to the main thread). Processes launched from such a thread inherit this signal mask which prevents them from properly shutting down.

Is this behaviour intentional? Wouldn't it make sense to clear the signal mask in the forked process?

**Comment by aleks-f** [comment]
The behavior of masking signals is obviously intentional. The behavior of the forked process is out of our control and you can do whatever you deem appropriate there. 

Forking from threads is a tricky business - in the child process, the forked-from thread becomes the main thread, while all other threads are dead.

**Reply by cgudrian** [reply]
Thanks for clarification! I indeed wasn't fully aware of the trickiness.
